katdal/chunkstore_s3.py

```diff
@@ -446,6 +465,15 @@ def create_array(self, array_name):
             with self.request(None, 'PUT', policy_url, data=json.dumps(policy)):
                 pass
 
+        if self.expiry_days > 0:
+            lifecycle_url = urllib.parse.urljoin(url, '?lifecycle')
```

Comment:
You can pass `params='lifecycle'` to `request` rather than mangling the url yourself.